packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java

```diff
@@ -0,0 +1,234 @@
+          Map<String, Object> configValues = new HashMap<>();
+          configValues.putAll(configProperties);
+          configValues.put("parameters", parseParameters(remoteConfig.getAll()));
+          return configValues;
+        });
+  }
+
+  private Map<String, Object> getConfigProperties(FirebaseRemoteConfig remoteConfig) {
+    Map<String, Object> configProperties = new HashMap<>();
+    configProperties.put(
+        "fetchTimeout", remoteConfig.getInfo().getConfigSettings().getFetchTimeoutInSeconds());
+    configProperties.put(
+        "minimumFetchInterval",
+        remoteConfig.getInfo().getConfigSettings().getMinimumFetchIntervalInSeconds());
+    configProperties.put("lastFetchTime", remoteConfig.getInfo().getFetchTimeMillis());
+    configProperties.put(
+        "lastFetchStatus", mapLastFetchStatus(remoteConfig.getInfo().getLastFetchStatus()));
+    Log.d(TAG, "Sending fetchTimeout: " + configProperties.get("fetchTimeout"));
+    return configProperties;
+  }
+
+  @Override
+  public Task<Void> didReinitializeFirebaseCore() {
+    return Tasks.call(() -> null);
+  }
+
+  private void setupChannel(BinaryMessenger messenger) {
+    registerPlugin(METHOD_CHANNEL, this);
+    channel = new MethodChannel(messenger, METHOD_CHANNEL);
+    channel.setMethodCallHandler(this);
+  }
+
+  private void tearDownChannel() {
+    channel.setMethodCallHandler(null);
+    channel = null;
+  }
+
+  private FirebaseRemoteConfig getRemoteConfig(Map<String, Object> arguments) {
+    String appName = (String) Objects.requireNonNull(arguments.get("appName"));
+    FirebaseApp app = FirebaseApp.getInstance(appName);
+    return FirebaseRemoteConfig.getInstance(app);
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, final MethodChannel.Result result) {
+    Task<?> methodCallTask;
+    FirebaseRemoteConfig remoteConfig = getRemoteConfig(call.arguments());
+
+    switch (call.method) {
+      case "RemoteConfig#ensureInitialized":
+        {
+          methodCallTask = remoteConfig.ensureInitialized();
+          break;
+        }
+      case "RemoteConfig#activate":
+        {
+          methodCallTask = remoteConfig.activate();
+          break;
+        }
+      case "RemoteConfig#getAll":
+        {
+          methodCallTask = Tasks.forResult(parseParameters(remoteConfig.getAll()));
+          break;
+        }
+      case "RemoteConfig#fetch":
+        {
+          methodCallTask = remoteConfig.fetch();
+          break;
+        }
+      case "RemoteConfig#fetchAndActivate":
+        {
+          methodCallTask = remoteConfig.fetchAndActivate();
+          break;
+        }
+      case "RemoteConfig#setConfigSettings":
+        {
+          int fetchTimeout = call.argument("fetchTimeout");
+          // To be insistent with iOS fetchTimeout is set to the default 1 minute (60 seconds)
+          // if an attempt is made to set it to zero.
+          if (fetchTimeout <= 0) {
```

Comment:
Would it make more sense to put this logic in the Dart code (in the public facing code), so `60` is passed to every platform rather than handling it just in Android?

If down the line another 3rd party platform like Windows was added and it has similar behaviour there is less chance of this being missed/inconsistent.

Reply:
Agreed, I added the check to the plugin level definition.